src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
+        if (directQuerySync === 'false') return false;
+      }
+      return null;
+    };
+
+    const shouldMountBanner = (): boolean => {
+      const urlFlag = getDirectQuerySyncFromUrl();
+      if (urlFlag !== null) {
+        return urlFlag;
+      }
+      return this.config.dashboardDirectQuerySyncEnabled;
+    };
+
+    const updateBanner = () => {
+      const appId = this.currentAppId;
+
+      if (appId === 'dashboards') {
```

Comment:
yeah we could maybe future proof this and make this configurable

Reply:
Transferring some of the discussion with @kavilla , I think for the current usage we may only consider the dashboard as the only scenerio for the only place to take this feature. Since the high level purpose of this feature is targeting on the dashboard only. But yes, I agree that I re name the feature flag to just `data_source_management.directQuerySyncEnabled`. So that we still leave the space for future configurations if we would like to have more use cases for this feature.